File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/businessmanagement/cs/winopportunity.cs
// <snippetwinopportunity>


 // Close the opportunity as won
 var winOppRequest = new WinOpportunityRequest
 {
     OpportunityClose = new OpportunityClose
     {
         OpportunityId = new EntityReference
             (Opportunity.EntityLogicalName, _opportunityId)
     },
     Status = new OptionSetValue((int)opportunity_statuscode.Won)
 };

 _serviceProxy.Execute(winOppRequest);

 Console.WriteLine("Opportunity closed as Won.");

// </snippetwinopportunity>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/businessmanagementvb/vb/winopportunity.vb
' <snippetwinopportunity>


 ' Close the opportunity as won
 Dim winOppRequest = New WinOpportunityRequest With
                     {
                         .OpportunityClose =
                         New OpportunityClose With
                         {
                             .OpportunityId =
                             New EntityReference(Opportunity.EntityLogicalName,
                                                 _opportunityId)},
                         .Status =
                         New OptionSetValue(
                             CInt(Fix(opportunity_statuscode.Won)))
                     }

 _serviceProxy.Execute(winOppRequest)

 Console.WriteLine("Opportunity closed as Won.")

' </snippetwinopportunity>